gossip/minimal/goshawk.go

```diff
@@ -83,22 +85,36 @@ func NewGoshawkFromFile(ctx context.Context, filename string, hc *http.Client, f
 
 // NewGoshawk creates a gossiper from the given configuration protobuf and optional http client.
 func NewGoshawk(ctx context.Context, cfg *configpb.GoshawkConfig, hc *http.Client, fetchOpts scanner.FetcherOptions) (*Goshawk, error) {
-	if cfg.DestHub == nil {
+	if len(cfg.DestHub) == 0 {
 		return nil, errors.New("no destination hub config found")
 	}
-	if cfg.SourceLog == nil || len(cfg.SourceLog) == 0 {
+	if len(cfg.SourceLog) == 0 {
 		return nil, errors.New("no source log config found")
 	}
 	if cfg.BufferSize < 0 {
 		return nil, fmt.Errorf("negative STH buffer size (%d) specified", cfg.BufferSize)
 	}
 
-	dest, err := hubScannerFromProto(cfg.DestHub, hc)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse dest hub config: %v", err)
+	hawk := Goshawk{
+		dests:     make(map[string]*hubScanner),
+		origins:   make(map[string]*originLog),
+		fetchOpts: fetchOpts,
+	}
+	hawk.fetchOpts.Continuous = true
+
+	for _, destHub := range cfg.DestHub {
```

Comment:
What about swapping the ordering of this block and the previous one so dests can just be set directly and dependencies are created first?

Reply:
I don't think that's straightforward because the `hubScanner` instances need to have a reference to the owning `Goshawk` (so they can find the per-source-log channel later).  But maybe I'm not quite following your suggestion?

Reply:
I might have missed the mutual dependency.